tests/test_jurisdiction_bulk_update.py

```diff
@@ -0,0 +1,64 @@
+from util.jurisdiction_bulk_update import bulk_update_jurisdictions
+from models import Election, Jurisdiction, JurisdictionAdministration, Organization, User
+import arlo_server
+import pytest
+import uuid
+
+
+@pytest.fixture
+def db():
+    with arlo_server.app.app_context():
+        arlo_server.init_db()
+
+    yield arlo_server.db
+
+    arlo_server.db.session.commit()
+    arlo_server.db.drop_all()
+    arlo_server.db.create_all()
+
+
+def test_first_update(db):
+    org = Organization(id=str(uuid.uuid4()), name='Test Org')
+    election = Election(id=str(uuid.uuid4()), organization=org)
+    new_admins = bulk_update_jurisdictions(db.session, election,
+                                           [('Jurisdiction #1', 'bob.harris@ca.gov')])
+
+    assert [(admin.jurisdiction.name, admin.user.email)
+            for admin in new_admins] == [('Jurisdiction #1', 'bob.harris@ca.gov')]
+
+    assert db.session.query(User).count() == 1
+    assert db.session.query(Jurisdiction).count() == 1
+    assert db.session.query(JurisdictionAdministration).count() == 1
+
+
+def test_idempotent(db):
+    org = Organization(id=str(uuid.uuid4()), name='Test Org')
+    election = Election(id=str(uuid.uuid4()), organization=org)
+
+    # Do it once.
+    bulk_update_jurisdictions(db.session, election, [('Jurisdiction #1', 'bob.harris@ca.gov')])
+
+    user = db.session.query(User).one()
+    jurisdiction = db.session.query(Jurisdiction).one()
+
+    # Do the same thing again.
+    bulk_update_jurisdictions(db.session, election, [('Jurisdiction #1', 'bob.harris@ca.gov')])
+
+    assert db.session.query(User).one() == user
+    assert db.session.query(Jurisdiction).one() == jurisdiction
+
+
+def test_remove_outdated_jurisdictions(db):
+    org = Organization(id=str(uuid.uuid4()), name='Test Org')
+    election = Election(id=str(uuid.uuid4()), organization=org)
+
+    # Add jurisdictions.
+    bulk_update_jurisdictions(db.session, election, [('Jurisdiction #1', 'bob.harris@ca.gov')])
```

Comment:
I'm assuming you purposely decided to not `commit` in `bulk_update_jurisdictions` so that transactional control can remain outside of the function call. That's cool. You might want to add a comment about that by the method definition.

Adding a comment here to ask: do you want to maybe `.commit()` after this call before finishing your test? Or are you explicitly trying to test things within the current transaction? I could see it either way, but for some reason am leaning towards committing before querying and asserting.

Reply:
> I'm assuming you purposely decided to not `commit` in `bulk_update_jurisdictions` so that transactional control can remain outside of the function call. That's cool. You might want to add a comment about that by the method definition.

That's true, though I realize now that I don't have a strong sense of how these sessions work. Is `db.session` always the default implicit session? Is `db.session` always the one used if you e.g. `User.query.count()`? Either way, my assertions always use the same session given to `bulk_update_jurisdictions` so it shouldn't require commiting before asserting. In my testing, probably the best reason to explicitly commit in the test before asserting is to trigger the SQL calls at a known point rather than implicitly elsewhere.

Reply:
Added a comment to the docstring and added `commit` calls in the tests.